### PR TITLE
better logging for weekly renewal

### PR DESCRIPTION
**Pull request by johnduffell**
This imrpoves the first iteration of the logging framework and adds some more logging to the renewal process.

Here are the relevant extracts from going to manage and renewing a weekly sub with a promo code in UAT

```
2017-03-08 11:01:37,306 model.SubscriptionOps$EnrichedPaperSubscription[ContextLogging.scala:14] INFO: {sub: A-S00068247} testing if renewable - wontAutoRenew: true, startedAlready: true, isOneOffPlan: true
2017-03-08 11:01:37,306 controllers.ManageWeekly$[ContextLogging.scala:14] INFO: {sub: A-S00068247} sub is renewable - showing weeklyRenew page
2017-03-08 11:02:04,048 model.SubscriptionOps$EnrichedPaperSubscription[ContextLogging.scala:14] INFO: {sub: A-S00068247} testing if renewable - wontAutoRenew: true, startedAlready: true, isOneOffPlan: true
2017-03-08 11:02:04,071 controllers.ManageWeekly$[ContextLogging.scala:14] INFO: {sub: A-S00068247} Attempting to renew Guardian Weekly Quarterly for Id(2c92c0f85aac6661015aad9389437b77) with promo code: Some(PromoCode(JACOB10))
2017-03-08 11:02:04,084 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} currentVersionExpired {false}
2017-03-08 11:02:04,085 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} startDateForRenewal {2018-03-01}
2017-03-08 11:02:04,405 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} zuoraAccount {2c92c0f859b047b70159bc8dcc901253}
2017-03-08 11:02:04,723 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} zuora bill to {2c92c0f859b047b70159bc8dcc971254}
2017-03-08 11:02:04,892 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} sfContact {003g000001LEkc7AAD}
2017-03-08 11:02:06,981 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} createPaymentMethod {2c92c0f859b047b70159bc8dcc901253}
2017-03-08 11:02:06,985 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} promo code from client {Some(PromoCode(JACOB10))}
2017-03-08 11:02:06,986 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} salesforce mailing country {Some(United Kingdom)}
2017-03-08 11:02:06,987 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} delivery country object {Some(Country(GB,United Kingdom,List()))}
2017-03-08 11:02:06,991 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: found promo {\/-(Promotion(4c565243-d6ac-4eeb-8fa3-59c0a427975e,JACOB10,10% Discount,AppliesTo(Set(ProductRatePlanId(2c92c0f9585841e7015862c9128e153b), ProductRatePlanId(2c92c0f958aa45650158da23e62d29e7), ProductRatePlanId(2c92c0f8574ebcdf015751506d7154ae), ProductRatePlanId(2c92c0f8574ebcdf015751506daf54c4), ProductRatePlanId(2c92c0f958aa45650158da23e5eb29d8), ProductRatePlanId(2c92c0f8574654af015747f934cc4a04), ProductRatePlanId(2c92c0f9574ee3d80157514ee1c36a8e), ProductRatePlanId(2c92c0f858aa38ab0158ac0204ce2b56), ProductRatePlanId(2c92c0f958aa45650158da23e5ab29c9)),Set(Country(SH,Saint Helena,List()), Country(IO,British Indian Ocean Territory,List()), Country(GP,Guadeloupe,List()), Country(VA,Holy See,List()), Country(KE,Kenya,List()), Country(MM,Myanmar,List()), Country(TL,East Timor,List()), Country(LV,Latvia,List()), Country(GL,Greenland,List()), Country(MV,Maldives,List()), Country(GT,Guatemala,List()), Country(GM,Gambia,List()), Country(GW,Guinea-Bissau,List()), Country(BT,Bhutan,List()), Country(SL,Sierra Leone,List()), Country(AO,Angola,List()), Country(RU,Russia,List()), Country(MQ,Martinique,List()), Country(VU,Vanuatu,List()), Country(GQ,Equatorial Guinea,List()), Country(PL,Poland,List()), Country(BG,Bulgaria,List()), Country(MR,Mauritania,List()), Country(AW,Aruba,List()), Country(TJ,Tajikistan,List()), Country(CK,Cook Islands,List()), Country(EC,Ecuador,List()), Country(BJ,Benin,List()), Country(KW,Kuwait,List()), Country(UA,Ukraine,List()), Country(FK,Falkland Islands,List()), Country(PS,Palestinian Territories,List()), Country(BH,Bahrain,List()), Country(AL,Albania,List()), Country(AI,Anguilla,List()), Country(TM,Turkmenistan,List()), Country(BM,Bermuda,List()), Country(ER,Eritrea,List()), Country(SB,Solomon Islands,List()), Country(SM,San Marino,List()), Country(SO,Somalia,List()), Country(PR,Puerto Rico,List()), Country(SI,Slovenia,List()), Country(QA,Qatar,List()), Country(PT,Portugal,List()), Country(WS,Samoa,List()), Country(BL,Saint Barthélemy,List()), Country(PF,French Polynesia,List()), Country(JO,Jordan,List()), Country(HK,Hong Kong,List()), Country(NA,Namibia,List()), Country(MZ,Mozambique,List()), Country(AS,American Samoa,List()), Country(ET,Ethiopia,List()), Country(KG,Kyrgyzstan,List()), Country(GG,Guernsey,List()), Country(LI,Liechtenstein,List()), Country(SG,Singapore,List()), Country(CY,Cyprus,List()), Country(MX,Mexico,List()), Country(LA,Laos,List()), Country(KY,Cayman Islands,List()), Country(SR,Suriname,List()), Country(BE,Belgium,List()), Country(DZ,Algeria,List()), Country(TK,Tokelau,List()), Country(HR,Croatia,List()), Country(AT,Austria,List()), Country(MC,Monaco,List()), Country(BN,Brunei Darussalam,List()), Country(AU,Australia,List()), Country(PK,Pakistan,List()), Country(BS,Bahamas,List()), Country(IS,Iceland,List()), Country(BZ,Belize,List()), Country(CR,Costa Rica,List()), Country(LT,Lithuania,List()), Country(NZ,New Zealand,List()), Country(SZ,Swaziland,List()), Country(LR,Liberia,List()), Country(US,United States,List(Alaska, Alabama, Arkansas, Arizona, California, Colorado, Connecticut, Delaware, Florida, Georgia, Guam, Hawaii, Iowa, Idaho, Illinois, Indiana, Kansas, Kentucky, Louisiana, Massachusetts, Maryland, Maine, Michigan, Minnesota, Missouri, Mississippi, Montana, North Carolina, North Dakota, Nebraska, New Hampshire, New Jersey, New Mexico, Nevada, New York, Ohio, Oklahoma, Oregon, Pennsylvania, Puerto Rico, Rhode Island, South Carolina, South Dakota, Tennessee, Texas, Utah, Virginia, Virgin Islands, Vermont, Washington, Washington DC, Wisconsin, West Virginia, Wyoming)), Country(CW,Curaçao,List()), Country(CL,Chile,List()), Country(RE,Réunion,List()), Country(RW,Rwanda,List()), Country(CN,China,List()), Country(NF,Norfolk Island,List()), Country(CZ,Czech Republic,List()), Country(EH,Western Sahara,List()), Country(PG,Papua New Guinea,List()), Country(IT,Italy,List()), Country(MF,Saint Martin,List()), Country(TR,Turkey,List()), Country(YT,Mayotte,List()), Country(KR,South Korea,List()), Country(TN,Tunisia,List()), Country(MA,Morocco,List()), Country(NL,Netherlands,List()), Country(JM,Jamaica,List()), Country(MP,Northern Mariana Islands,List()), Country(IR,Iran,List()), Country(AQ,Antarctica,List()), Country(HM,Heard Island and McDonald Islands,List()), Country(UM,United States Minor Outlying Islands,List()), Country(MH,Marshall Islands,List()), Country(CD,Congo (Kinshasa),List()), Country(BY,Belarus,List()), Country(ZA,South Africa,List()), Country(TO,Tonga,List()), Country(SY,Syria,List()), Country(IN,India,List()), Country(TD,Chad,List()), Country(DK,Denmark,List()), Country(CM,Cameroon,List()), Country(DM,Dominica,List()), Country(CX,Christmas Island,List()), Country(SK,Slovakia,List()), Country(BV,Bouvet Island,List()), Country(BW,Botswana,List()), Country(ES,Spain,List()), Country(EG,Egypt,List()), Country(AG,Antigua & Barbuda,List()), Country(TH,Thailand,List()), Country(MD,Moldova,List()), Country(OM,Oman,List()), Country(VC,Saint Vincent & The Grenadines,List()), Country(SA,Saudi Arabia,List()), Country(CI,Ivory Coast,List()), Country(CO,Colombia,List()), Country(AD,Andorra,List()), Country(TF,French Southern Territories,List()), Country(KH,Cambodia,List()), Country(SN,Senegal,List()), Country(PM,Saint Pierre & Miquelon,List()), Country(CF,Central African Republic,List()), Country(NE,Niger,List()), Country(PN,Pitcairn Islands,List()), Country(CU,Cuba,List()), Country(FM,Micronesia,List()), Country(HT,Haiti,List()), Country(TZ,Tanzania,List()), Country(VE,Venezuela,List()), Country(MO,Macau,List()), Country(BF,Burkina Faso,List()), Country(CA,Canada,List(Alberta, British Columbia, Manitoba, New Brunswick, Newfoundland and Labrador, Nova Scotia, Northwest Territories, Nunavut, Ontario, Prince Edward Island, Quebec, Saskatchewan, Yukon)), Country(MG,Madagascar,List()), Country(ME,Montenegro,List()), Country(BD,Bangladesh,List()), Country(CG,Congo (Brazzaville),List()), Country(JE,Jersey,List()), Country(MY,Malaysia,List()), Country(TG,Togo,List()), Country(BA,Bosnia-Herzegovina,List()), Country(HU,Hungary,List()), Country(KI,Kiribati,List()), Country(UZ,Uzbekistan,List()), Country(MW,Malawi,List()), Country(FR,France,List()), Country(AZ,Azerbaijan,List()), Country(PW,Palau,List()), Country(KP,North Korea,List()), Country(BO,Bolivia,List()), Country(ST,Sao Tome & Principe,List()), Country(IM,Isle of Man,List()), Country(AF,Afghanistan,List()), Country(BB,Barbados,List()), Country(KZ,Kazakhstan,List()), Country(MT,Malta,List()), Country(IL,Israel,List()), Country(MS,Montserrat,List()), Country(PH,Philippines,List()), Country(LK,Sri Lanka,List()), Country(DJ,Djibouti,List()), Country(SC,Seychelles,List()), Country(DO,Dominican Republic,List()), Country(YE,Yemen,List()), Country(ZW,Zimbabwe,List()), Country(VG,British Virgin Islands,List()), Country(GI,Gibraltar,List()), Country(LY,Libya,List()), Country(BQ,Bonaire, Sint Eustatius and Saba,List()), Country(TC,Turks & Caicos Islands,List()), Country(SJ,Svalbard and Jan Mayen,List()), Country(IE,Ireland,List()), Country(CV,Cape Verde Islands,List()), Country(KN,Saint Christopher & Nevis,List()), Country(DE,Germany,List()), Country(FI,Finland,List()), Country(GN,Guinea,List()), Country(IQ,Iraq,List()), Country(GA,Gabon,List()), Country(JP,Japan,List()), Country(AR,Argentina,List()), Country(PE,Peru,List()), Country(FJ,Fiji,List()), Country(TW,Taiwan,List()), Country(GE,Georgia,List()), Country(HN,Honduras,List()), Country(UG,Uganda,List()), Country(LB,Lebanon,List()), Country(ML,Mali,List()), Country(RS,Serbia,List()), Country(BR,Brazil,List()), Country(GR,Greece,List()), Country(GU,Guam,List()), Country(CC,Cocos (Keeling) Islands,List()), Country(NU,Niue,List()), Country(VI,United States Virgin Islands,List()), Country(GH,Ghana,List()), Country(BI,Burundi,List()), Country(GF,French Guiana,List()), Country(TV,Tuvalu,List()), Country(AE,United Arab Emirates,List()), Country(GY,Guyana,List()), Country(GB,United Kingdom,List()), Country(EE,Estonia,List()), Country(ID,Indonesia,List()), Country(LS,Lesotho,List()), Country(PA,Panama,List()), Country(WF,Wallis & Futuna,List()), Country(CH,Switzerland,List()), Country(PY,Paraguay,List()), Country(ZM,Zambia,List()), Country(GD,Grenada,List()), Country(LU,Luxembourg,List()), Country(SV,El Salvador,List()), Country(UY,Uruguay,List()), Country(AM,Armenia,List()), Country(GS,South Georgia & The South Sandwish Islands,List()), Country(NP,Nepal,List()), Country(KM,Comoros,List()), Country(MN,Mongolia,List()), Country(TT,Trinidad & Tobago,List()), Country(NR,Nauru,List()), Country(RO,Romania,List()), Country(LC,Saint Lucia,List()), Country(MK,Macedonia,List()), Country(NO,Norway,List()), Country(SE,Sweden,List()), Country(FO,Faeroe Islands,List()), Country(SD,Sudan,List()), Country(AX,Åland Islands,List()), Country(VN,Vietnam,List()), Country(NC,New Caledonia,List()), Country(NI,Nicaragua,List()), Country(MU,Mauritius,List()), Country(SX,Sint Maarten,List()), Country(NG,Nigeria,List()))),CampaignCode(C_IVWLO9GR),Map(Channel(Channel 0) -> Set(PromoCode(JACOB10))),None,2017-01-01T00:00:00.000Z,Some(2019-01-01T23:59:59.000Z),PercentDiscount))}
2017-03-08 11:02:06,992 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: upgrade/renewal for context {\/-(PercentDiscount)}
2017-03-08 11:02:06,993 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: effective promotion type {\/-(PercentDiscount)}
2017-03-08 11:02:06,994 com.gu.memsub.promo.LogImplicit$[LogImplicit.scala:9] INFO: validated promo type {\/-(PercentDiscount)}
2017-03-08 11:02:06,995 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} validated promotion {\/-(ValidPromotion(PromoCode(JACOB10),Promotion(4c565243-d6ac-4eeb-8fa3-59c0a427975e,JACOB10,10% Discount,AppliesTo(Set(ProductRatePlanId(2c92c0f9585841e7015862c9128e153b), ProductRatePlanId(2c92c0f958aa45650158da23e62d29e7), ProductRatePlanId(2c92c0f8574ebcdf015751506d7154ae), ProductRatePlanId(2c92c0f8574ebcdf015751506daf54c4), ProductRatePlanId(2c92c0f958aa45650158da23e5eb29d8), ProductRatePlanId(2c92c0f8574654af015747f934cc4a04), ProductRatePlanId(2c92c0f9574ee3d80157514ee1c36a8e), ProductRatePlanId(2c92c0f858aa38ab0158ac0204ce2b56), ProductRatePlanId(2c92c0f958aa45650158da23e5ab29c9)),Set(Country(SH,Saint Helena,List()), Country(IO,British Indian Ocean Territory,List()), Country(GP,Guadeloupe,List()), Country(VA,Holy See,List()), Country(KE,Kenya,List()), Country(MM,Myanmar,List()), Country(TL,East Timor,List()), Country(LV,Latvia,List()), Country(GL,Greenland,List()), Country(MV,Maldives,List()), Country(GT,Guatemala,List()), Country(GM,Gambia,List()), Country(GW,Guinea-Bissau,List()), Country(BT,Bhutan,List()), Country(SL,Sierra Leone,List()), Country(AO,Angola,List()), Country(RU,Russia,List()), Country(MQ,Martinique,List()), Country(VU,Vanuatu,List()), Country(GQ,Equatorial Guinea,List()), Country(PL,Poland,List()), Country(BG,Bulgaria,List()), Country(MR,Mauritania,List()), Country(AW,Aruba,List()), Country(TJ,Tajikistan,List()), Country(CK,Cook Islands,List()), Country(EC,Ecuador,List()), Country(BJ,Benin,List()), Country(KW,Kuwait,List()), Country(UA,Ukraine,List()), Country(FK,Falkland Islands,List()), Country(PS,Palestinian Territories,List()), Country(BH,Bahrain,List()), Country(AL,Albania,List()), Country(AI,Anguilla,List()), Country(TM,Turkmenistan,List()), Country(BM,Bermuda,List()), Country(ER,Eritrea,List()), Country(SB,Solomon Islands,List()), Country(SM,San Marino,List()), Country(SO,Somalia,List()), Country(PR,Puerto Rico,List()), Country(SI,Slovenia,List()), Country(QA,Qatar,List()), Country(PT,Portugal,List()), Country(WS,Samoa,List()), Country(BL,Saint Barthélemy,List()), Country(PF,French Polynesia,List()), Country(JO,Jordan,List()), Country(HK,Hong Kong,List()), Country(NA,Namibia,List()), Country(MZ,Mozambique,List()), Country(AS,American Samoa,List()), Country(ET,Ethiopia,List()), Country(KG,Kyrgyzstan,List()), Country(GG,Guernsey,List()), Country(LI,Liechtenstein,List()), Country(SG,Singapore,List()), Country(CY,Cyprus,List()), Country(MX,Mexico,List()), Country(LA,Laos,List()), Country(KY,Cayman Islands,List()), Country(SR,Suriname,List()), Country(BE,Belgium,List()), Country(DZ,Algeria,List()), Country(TK,Tokelau,List()), Country(HR,Croatia,List()), Country(AT,Austria,List()), Country(MC,Monaco,List()), Country(BN,Brunei Darussalam,List()), Country(AU,Australia,List()), Country(PK,Pakistan,List()), Country(BS,Bahamas,List()), Country(IS,Iceland,List()), Country(BZ,Belize,List()), Country(CR,Costa Rica,List()), Country(LT,Lithuania,List()), Country(NZ,New Zealand,List()), Country(SZ,Swaziland,List()), Country(LR,Liberia,List()), Country(US,United States,List(Alaska, Alabama, Arkansas, Arizona, California, Colorado, Connecticut, Delaware, Florida, Georgia, Guam, Hawaii, Iowa, Idaho, Illinois, Indiana, Kansas, Kentucky, Louisiana, Massachusetts, Maryland, Maine, Michigan, Minnesota, Missouri, Mississippi, Montana, North Carolina, North Dakota, Nebraska, New Hampshire, New Jersey, New Mexico, Nevada, New York, Ohio, Oklahoma, Oregon, Pennsylvania, Puerto Rico, Rhode Island, South Carolina, South Dakota, Tennessee, Texas, Utah, Virginia, Virgin Islands, Vermont, Washington, Washington DC, Wisconsin, West Virginia, Wyoming)), Country(CW,Curaçao,List()), Country(CL,Chile,List()), Country(RE,Réunion,List()), Country(RW,Rwanda,List()), Country(CN,China,List()), Country(NF,Norfolk Island,List()), Country(CZ,Czech Republic,List()), Country(EH,Western Sahara,List()), Country(PG,Papua New Guinea,List()), Country(IT,Italy,List()), Country(MF,Saint Martin,List()), Country(TR,Turkey,List()), Country(YT,Mayotte,List()), Country(KR,South Korea,List()), Country(TN,Tunisia,List()), Country(MA,Morocco,List()), Country(NL,Netherlands,List()), Country(JM,Jamaica,List()), Country(MP,Northern Mariana Islands,List()), Country(IR,Iran,List()), Country(AQ,Antarctica,List()), Country(HM,Heard Island and McDonald Islands,List()), Country(UM,United States Minor Outlying Islands,List()), Country(MH,Marshall Islands,List()), Country(CD,Congo (Kinshasa),List()), Country(BY,Belarus,List()), Country(ZA,South Africa,List()), Country(TO,Tonga,List()), Country(SY,Syria,List()), Country(IN,India,List()), Country(TD,Chad,List()), Country(DK,Denmark,List()), Country(CM,Cameroon,List()), Country(DM,Dominica,List()), Country(CX,Christmas Island,List()), Country(SK,Slovakia,List()), Country(BV,Bouvet Island,List()), Country(BW,Botswana,List()), Country(ES,Spain,List()), Country(EG,Egypt,List()), Country(AG,Antigua & Barbuda,List()), Country(TH,Thailand,List()), Country(MD,Moldova,List()), Country(OM,Oman,List()), Country(VC,Saint Vincent & The Grenadines,List()), Country(SA,Saudi Arabia,List()), Country(CI,Ivory Coast,List()), Country(CO,Colombia,List()), Country(AD,Andorra,List()), Country(TF,French Southern Territories,List()), Country(KH,Cambodia,List()), Country(SN,Senegal,List()), Country(PM,Saint Pierre & Miquelon,List()), Country(CF,Central African Republic,List()), Country(NE,Niger,List()), Country(PN,Pitcairn Islands,List()), Country(CU,Cuba,List()), Country(FM,Micronesia,List()), Country(HT,Haiti,List()), Country(TZ,Tanzania,List()), Country(VE,Venezuela,List()), Country(MO,Macau,List()), Country(BF,Burkina Faso,List()), Country(CA,Canada,List(Alberta, British Columbia, Manitoba, New Brunswick, Newfoundland and Labrador, Nova Scotia, Northwest Territories, Nunavut, Ontario, Prince Edward Island, Quebec, Saskatchewan, Yukon)), Country(MG,Madagascar,List()), Country(ME,Montenegro,List()), Country(BD,Bangladesh,List()), Country(CG,Congo (Brazzaville),List()), Country(JE,Jersey,List()), Country(MY,Malaysia,List()), Country(TG,Togo,List()), Country(BA,Bosnia-Herzegovina,List()), Country(HU,Hungary,List()), Country(KI,Kiribati,List()), Country(UZ,Uzbekistan,List()), Country(MW,Malawi,List()), Country(FR,France,List()), Country(AZ,Azerbaijan,List()), Country(PW,Palau,List()), Country(KP,North Korea,List()), Country(BO,Bolivia,List()), Country(ST,Sao Tome & Principe,List()), Country(IM,Isle of Man,List()), Country(AF,Afghanistan,List()), Country(BB,Barbados,List()), Country(KZ,Kazakhstan,List()), Country(MT,Malta,List()), Country(IL,Israel,List()), Country(MS,Montserrat,List()), Country(PH,Philippines,List()), Country(LK,Sri Lanka,List()), Country(DJ,Djibouti,List()), Country(SC,Seychelles,List()), Country(DO,Dominican Republic,List()), Country(YE,Yemen,List()), Country(ZW,Zimbabwe,List()), Country(VG,British Virgin Islands,List()), Country(GI,Gibraltar,List()), Country(LY,Libya,List()), Country(BQ,Bonaire, Sint Eustatius and Saba,List()), Country(TC,Turks & Caicos Islands,List()), Country(SJ,Svalbard and Jan Mayen,List()), Country(IE,Ireland,List()), Country(CV,Cape Verde Islands,List()), Country(KN,Saint Christopher & Nevis,List()), Country(DE,Germany,List()), Country(FI,Finland,List()), Country(GN,Guinea,List()), Country(IQ,Iraq,List()), Country(GA,Gabon,List()), Country(JP,Japan,List()), Country(AR,Argentina,List()), Country(PE,Peru,List()), Country(FJ,Fiji,List()), Country(TW,Taiwan,List()), Country(GE,Georgia,List()), Country(HN,Honduras,List()), Country(UG,Uganda,List()), Country(LB,Lebanon,List()), Country(ML,Mali,List()), Country(RS,Serbia,List()), Country(BR,Brazil,List()), Country(GR,Greece,List()), Country(GU,Guam,List()), Country(CC,Cocos (Keeling) Islands,List()), Country(NU,Niue,List()), Country(VI,United States Virgin Islands,List()), Country(GH,Ghana,List()), Country(BI,Burundi,List()), Country(GF,French Guiana,List()), Country(TV,Tuvalu,List()), Country(AE,United Arab Emirates,List()), Country(GY,Guyana,List()), Country(GB,United Kingdom,List()), Country(EE,Estonia,List()), Country(ID,Indonesia,List()), Country(LS,Lesotho,List()), Country(PA,Panama,List()), Country(WF,Wallis & Futuna,List()), Country(CH,Switzerland,List()), Country(PY,Paraguay,List()), Country(ZM,Zambia,List()), Country(GD,Grenada,List()), Country(LU,Luxembourg,List()), Country(SV,El Salvador,List()), Country(UY,Uruguay,List()), Country(AM,Armenia,List()), Country(GS,South Georgia & The South Sandwish Islands,List()), Country(NP,Nepal,List()), Country(KM,Comoros,List()), Country(MN,Mongolia,List()), Country(TT,Trinidad & Tobago,List()), Country(NR,Nauru,List()), Country(RO,Romania,List()), Country(LC,Saint Lucia,List()), Country(MK,Macedonia,List()), Country(NO,Norway,List()), Country(SE,Sweden,List()), Country(FO,Faeroe Islands,List()), Country(SD,Sudan,List()), Country(AX,Åland Islands,List()), Country(VN,Vietnam,List()), Country(NC,New Caledonia,List()), Country(NI,Nicaragua,List()), Country(MU,Mauritius,List()), Country(SX,Sint Maarten,List()), Country(NG,Nigeria,List()))),CampaignCode(C_IVWLO9GR),Map(Channel(Channel 0) -> Set(PromoCode(JACOB10))),None,2017-01-01T00:00:00.000Z,Some(2019-01-01T23:59:59.000Z),PercentDiscount)))}
2017-03-08 11:02:10,189 services.CheckoutService[ContextLogging.scala:14] INFO: {sub: A-S00068247} addPlan {List(2c92c0f95aac74ff015aad954ec07b71, 2c92c0f95aac74ff015aad954f647b7f, 2c92c0f95aac74ff015aad9550087b91, 2c92c0f95aac74ff015aad9550f47ba4)}
2017-03-08 11:02:11,382 com.gu.zuora.ZuoraService[ZuoraService.scala:136] INFO: updated promo code
2017-03-08 11:02:11,772 services.ExactTargetService[ContextLogging.scala:14] INFO: {sub: A-S00068247} Successfully enqueued A-S00068247's guardian weekly renewal email.
```

It's not perfect, but it's better!  And can and should be improved later.  @pvighi @paulbrown1982 @jacobwinch

### CODE REVIEW

File: app/services/ExactTargetService.scala [hunk 1]
```diff
@@ -214,12 +216,16 @@ class ExactTargetService(
       val code = "ET002"
     }
 
-    def getPaymentMethod(accountId: AccountId): Future[\/[Error, PaymentMethod]] = paymentService.getPaymentMethod(oldSub.accountId).map(_.toRightDisjunction(SimpleError(s"Failed to enqueue ${oldSub.name.get}'s guardian weekly renewal email. No payment method found in account")))
+    def getPaymentMethod(accountId: AccountId): Future[\/[Error, PaymentMethod]] =
```
Comment (johnduffell) [inline app/services/ExactTargetService.scala]: mostly just tweaking the formatting a bit, the logging was already good here

**Comment by pvighi** [comment]
I don't think this is introduced in this PR but
in lines for 2017-03-08 11:02:06,991 and 2017-03-08 11:02:06,995 in your log example, do we need to log all the promotion info like that?

**Comment by jacobwinch** [comment]
👍 

I agree that we should avoid logging the whole country list - very noisy when trying to debug - but feel free to merge this PR and we can remove that as a separate task.

**Comment by johnduffell** [comment]
yeah also agreed on the countries, thanks for the pointer, we can deal with that in future I think

File: app/logging/ContextLogging.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  def error[P <: AnyPlan : Subscription](message: String): Unit = {
+    val context = implicitly[Subscription[P]]
+    logger.info(s"{sub: ${context.name.get}} $message")
+  }
+
+  implicit class FutureContextLoggable[T](future: Future[T]) {
+    def withContextLogging[P <: AnyPlan : Subscription](message: String, extractor: T => Any = identity): Future[T] = {
+      future.foreach(any => info(s"$message {${extractor(any)}}"))
```
Comment (johnduffell) [inline app/logging/ContextLogging.scala]: I should have made this onComplete and logged failure as well as success 🍌

**Pull request merged**